rascaline-torch/include/rascaline/torch/calculator.hpp

```diff
@@ -15,6 +15,60 @@ class RascalineAutograd;
 class CalculatorHolder;
 using TorchCalculator = torch::intrusive_ptr<CalculatorHolder>;
 
+class CalculatorOptionsHolder;
+using TorchCalculatorOptions = torch::intrusive_ptr<CalculatorOptionsHolder>;
+
+/// Options for a single calculation
+class RASCALINE_TORCH_EXPORT CalculatorOptionsHolder: public torch::CustomClassHolder {
```

Comment:
sorry could not figure what this syntax is doing. What is this macro doing to the class declaration?

Reply:
This is originally a Windows thing, where you need to explicitly declare which classes/functions should be exported & visible from a DLL/shared library. We also add a flag to mimic this behavior on Linux/macOS (`-fvisibility=hidden`).

The macro is generated by CMake (check `target/tmp/torch-tests/include/rascaline/torch/exports.h` for what it looks like on your system) to expand to the right attribute both when compiling the code & when using the code from an external project.

Reply:
okay I kind of get this